dashboard/app/controllers/zendesk_session_controller.rb

```diff
@@ -9,15 +9,29 @@ class ZendeskSessionController < ApplicationController
   SECRET = Dashboard::Application.config.zendesk_secret
   SUBDOMAIN = Dashboard::Application.config.zendesk_subdomain
 
+  @error = nil
+
   def index
-    unless current_user.age && current_user.under_13?
+    if allow_zendesk_signin?(current_user)
       sign_into_zendesk(current_user)
       return
     end
   end
 
   private
 
+  def allow_zendesk_signin?(user)
+    if user.age && user.under_13?
+      @error = I18n.t('zendesk_too_young_message', support_url: "support.code.org", markdown: true)
+      return false
+    end
+    if user.email && Mail::Address.new(user.email).domain == "code.org" && !user.google_oauth_email_matches?
```

Comment:
This would be a good spot for a comment explaining _why_ we restrict Zendesk SSO for code.org accounts that don't have matching Google OAuth configured.